### ISSUE TITLE
Remove dependency on Security.Cryptography

### ISSUE THREAD

**Issue by MiYanni** [Azure/azure-powershell#5756]
In the project folder, the Utilities folder contains a dll, `Security.Cryptography.dll`. The code for this library is [no longer maintained](https://archive.codeplex.com/?p=clrsecurity). The library itself was added to [NuGet](https://www.nuget.org/packages/Security.Cryptography/) using the last version that was created. That version is 1.7.2. The dll in the Utilities folder is 1.6.0.

#### Projects

- AzureBackup
- RecoveryServices
- RecoveryServices.SiteRecovery
- SiteRecovery

#### Work Items

- [ ] Remove the checked-in dll and remove the dependency in the .NET Framework project
  - Suggestion is to use what is available in `System.Security.Cryptography`
- [ ] Fix the files affected by this
  - `CertUtils.cs` is in all the listed projects
  - Usages are `X509CertificateCreationParameters`, `CreateSelfSignedCertificate`, and `CngAlgorithm2` 
- [ ] Validate both .NET Core and .NET Framework projects build properly

**Note**: I cannot reference 1.7.2 as it is not strong-name signed. I will continue to use 1.6.0 in NetCore until this issue is resolved. Referencing a non-NetCore dll is a [newer feature](https://stackoverflow.com/a/44999938/294804) of .NET Core.

**Comment by MiYanni** [comment]
I spent the last 2 days looking at solutions for this. Unfortunately, self-signed certificates are OS dependent. For Windows, it using the Windows API, [CertEnroll](https://msdn.microsoft.com/en-us/library/windows/desktop/aa374863(v=vs.85).aspx). The only implementations I've found for C#/PowerShell using CertEnroll ([here](https://gist.github.com/jimmyca15/8f737f5f0bcf347450bd6d6bf34f4f7e) and [here](https://github.com/Crypt32/PSPKI/blob/master/PSPKI/Client/New-SelfSignedCertificateEx.ps1)). This will not work for .NET Core as they are not cross-platform.

The only alternative I can think of is using a separate method of auth to the RecoveryService REST API. I'm not sure if it exists. But, using self-signed certs cannot be made cross-platform easily without significant work.
